File: src/oracle_todo/paths.py
from __future__ import annotations

import os
from pathlib import Path


def todo_home() -> Path:
    return Path(os.environ.get("ORACLE_TODO_HOME", "~/.hermes/oracle-todo")).expanduser()


def db_path() -> Path:
    return todo_home() / "todo.sqlite"


def exports_dir() -> Path:
    return todo_home() / "exports"


def events_jsonl_path() -> Path:
    return todo_home() / "events.jsonl"

File: src/infrastructure/paths.rs
use std::path::{Path, PathBuf};

use anyhow::{Result, anyhow};

pub fn todo_home(explicit_home: Option<PathBuf>) -> Result<PathBuf> {
    if let Some(home) = explicit_home {
        return Ok(home);
    }
    if let Some(home) = std::env::var_os("ORACLE_TODO_HOME") {
        return Ok(PathBuf::from(home));
    }
    let home = std::env::var_os("HOME").ok_or_else(|| anyhow!("HOME is not set"))?;
    Ok(PathBuf::from(home).join(".hermes/oracle-todo"))
}

pub fn db_path(home: &Path) -> PathBuf {
    home.join("todo.sqlite")
}

pub fn exports_dir(home: &Path) -> PathBuf {
    home.join("exports")
}
